README.md

```diff
@@ -1 +1,401 @@
+            functions: 80,
+            lines: 80,
+            statements: -10
+        })
+        .tap('global', (thresholds) => Object.assign(thresholds, { branches: 100 }))
+        .end()
+    .rootDir('my-project')
+    .runner('jest-runner-mocha');
+```
+
+
+### OrderableChainedMap
+
+OrderableChainedMap extends [Chainable](#chained-map) and allows to re-order keys via the `order(key, specifier)` method. It also adds `before(key)` and `after(key)` methods on Chainables added via `set`. This allows an entry to declare that it comes before or after another by calling the aforementioned methods on itself.
+
+Consequently, `keys()`, `values()`, `entries()` and `.forEach()` methods of OrderableChainedMap will have into consideration any calls changes to the entries order.
+
+<details>
+  <summary><b>Diferences from webpack-chain:</b></summary>
+  <ul>
+    <li>`webpack-chain` has Orderable which is a function that receives a Chainable and adds the `before` and `after` methods. OrderableChainedMap is more flexible since it allows ordering entries whose values are primitives</li>
+    <li>Removed weird treatment on <code>.merge(obj, [omit])</code> of <code>after</code> and <code>before</code> keys in <code>obj</code></li>
+    <li>Fixed minor bugs that caused <code>.keys()</code>, <code>.values()</code> and <code>.entries()</code> to not respect the order specified with <code>.before(key)</code> and <code>.after(key)</code>
+  </ul>
+</details>
+
+
+#### .order(key, specifier)
+
+Sets the order of the entry corresponding to `key` relative to another specified by `specifier`.
+
+`specifier` gets called with a single argument which is an object with `before(relativeKey)` and `after(relativeKey)` functions.
+
+Returns itself to allow chaining.
+
+
+#### Methods added to Chainables included via `set`
+
+##### .before(key)
+
+Marks the entry to be positioned before `key` on the OrderableChainedMap.
+
+Returns itself to allow chaining.
+
+##### .after(key)
+
+Marks the entry to be positioned after `key` on the OrderableChainedMap.
+
+Returns itself to allow chaining.
+
+
+#### Example
+
+```js
+class Jest extends ChainedMap {
+    constructor(parent) {
+        super(parent);
+
+        this.extend([
+            'rootDir',
+            'runner',
+            // ...
+        ]);
+
+        this.setupFiles = new OrderableChainedMap(this);
+        this.projects = new OrderableChainedMap(this);
+    }
+
+    project(name) {
+        if (!this.projects.has(name)) {
+            this.projects.set(name, new ChainedMap(this));
+        }
+
+        return this.projects.get(name);
+    }
+}
+
+// Then use it like so:
+const jestConfig = new Jest();
+
+jestConfig
+    // Using `order` to re-order entries
+    .setupFiles
+        .set('setup-foo', '/path/to/setup/foo.js')
+        .set('setup-bar', '/path/to/setup/bar.js')
+        .set('setup-baz', '/path/to/setup/baz.js')
+        .order('setup-bar', ({ before }) => before('setup-foo'))
```

Comment:
I ended up with this signature because it's more semantic, but here's a few alternatives:

`.orderAfter('setup-bar', 'setup-foo')` / .orderBefore('setup-bar', 'setup-foo')` - not every obvious which one is after what
`order('setup-bar', 'before', 'setup-foo')`
`order('setup-bar', { relativeTo: 'setup-foo', position: 'after' })`

I prefer the current approach because it's more description, but I would like your opinion.

Reply:
I like this approach to 👍

What do you think of renaming it to `reorder` or `move`?

Reply:
I like both actually! Lets wait for @acostalima and @andregoncalvesdev feedback on their preference.

Reply:
I've talked with @andregoncalvesdev and we settled on `move`.